lmfdb/utils/web_display.py

```diff
@@ -783,3 +784,14 @@ def list_to_latex_matrix(li):
     mm += r"\\".join(" & ".join(str(a) for a in row) for row in li)
     mm += r'\end{array}\right)'
     return mm
+
+
+def dispZmat_from_list(a_list, my_dim):
+    r"""Display a matrix with integer entries from a list
+    """
+    num_entries = len(a_list)
+    assert num_entries == my_dim ** 2
+    output = []
+    for i in range(my_dim):
```

Comment:
This won't work.  You need `range(0,my_dim**2,my_dim)`.  I might also just call it `dim` rather than `my_dim` for simplicity.

Reply:
oops ... yes indeed, sorted now